src/modules/ekf2/EKF/ekf.cpp

```diff
@@ -280,8 +280,7 @@ bool Ekf::resetGlobalPosToExternalObservation(double lat_deg, double lon_deg, fl
 		uint64_t timestamp_observation)
 {
 	if (!_pos_ref.isInitialized()) {
-		ECL_WARN("unable to reset global position, position reference not initialized");
-		return false;
+		return setEkfGlobalOriginFromCurrentPos(lat_deg, lon_deg, NAN, accuracy, NAN);
```

Comment:
A signature like this is a bit awkward when you only have part of the data, why not use the helper for what you actually have? Then `setEkfGlobalOriginFromCurrentPos()` could actually succeed or fail atomically for all of lat/lon/alt.

```suggestion
		return setLatLonOriginFromCurrentPos(lat_deg, lon_deg, accuracy);
```